spicy/toolchain/src/compiler/codegen/parser-builder.cc

```diff
@@ -277,6 +277,13 @@ struct ProductionVisitor : public production::Visitor {
                         pushState(std::move(pstate));
                         pb->initializeUnit(p.location());
                     }
+
+                    if ( p.meta().field() && ! p.meta().field()->isTransient() && addl_param ) {
+                        if ( auto a = p.meta().field()->attributes()->find(hilti::attribute::Kind::Size);
```

Comment:
Do you really want `&size` here? That's the number of bytes that the vector is being parsed from. For `uint8` elements, that happens to be the right number, but for like `uint16` it wouldn't be. 

It seems that the optimization should instead be applied to cases like `: uint8[1000]` where we have the actual vector size in terms of number of elements?

Reply:
Oops, thanks for pointing that out, I considered it then wrongly dismissed it when implementing

I'll see what I can do for this, anything with a known size should be possible, but for units it may get tricky. I'd rather not add in too much logic for this too, though, since it seems like it'd mostly be a relatively small difference for normal sized vectors and it'd be more maintenance.

Reply:
I just ditched the `&size` part, it's too much to try to guess with skip fields and whatnot in order to correlate to the number of units, for example. Though you probably already came to that conclusion